app/models/appdata.rb

```diff
@@ -5,53 +5,64 @@
 require 'zlib'
 
 class Appdata
-  def self.get(dist = 'factory')
-    data = {}
-    xml = Appdata.get_distribution(dist, 'oss')
-    data = add_appdata(data, xml)
-    xml = Appdata.get_distribution(dist, 'non-oss')
-    data = add_appdata(data, xml)
-    data
+  def initialize(distribution)
+    @distribution = distribution
+    @data = nil
+  end
+
+  def data
```

Comment:
not sure if we need to load this dynamically. I assume that always use data when creating an Appdata instance.

Reply:
Right, I'll move the loading to `initialize`